Moya.swift

```diff
@@ -126,11 +126,13 @@ public class MoyaProvider<T: MoyaTarget> {
             let behavior = stubBehavior(token)
 
             let stub: () -> () = {
-                switch endpoint.sampleResponse {
+                switch endpoint.sampleResponse.evaluate() {
                 case .Success(let statusCode, let data):
                     completion(data: data, statusCode: statusCode, response:nil, error: nil)
                 case .Error(let statusCode, let error, let data):
                     completion(data: data, statusCode: statusCode, response:nil, error: error)
+                default:
```

Comment:
Hmm. Maybe we can rethink this a bit. I'm really OK with dropping support for all non-lazy sample responses. Maybe instead of adding to the enum, we can replace the `sampleResponse` property on `Endpoint` with a closure that returns a sample response. Thoughts?